pkg/rng/entropy.go

```diff
@@ -0,0 +1,139 @@
+package rng
+
+import (
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var (
+	trngList map[string]int
+)
+
+const (
+	hwRandomDir                string        = "/sys/class/misc/hw_random"
+	hwRandomCurrentFile        string        = "/sys/class/misc/hw_random/rng_current"
+	hwRandomAvailableFile      string        = "/sys/class/misc/hw_random/rng_available"
+	randomPoolSizeFile         string        = "/proc/sys/kernel/random/poolsize"
+	randomEntropyAvailableFile string        = "/proc/sys/kernel/random/entropy_avail"
+	entropyFeedTime            time.Duration = 10
+	randomDevice               string        = "/dev/random"
+	hwRandomDevice             string        = "/dev/hwrng"
+)
+
+func init() {
+	trngList = make(map[string]int)
+
+	// Priorize the HW RNG 0 is the highest
+	trngList["tpm-rng"] = 0
+	trngList["intel-rng"] = 0
+	trngList["amd-rng"] = 0
+	trngList["timeriomem-rng"] = 3
+}
+
+func setAvailableTRNG() (bool, error) {
```

Comment:
maybe a brief comment on the function? It's not too clear what it does and what it returns. It becomes clear to me only after reading it all, and after reading `UpdateLinuxRandomness`